packages/teleport/src/components/TextSelectCopy/TextSelectCopyMulti.tsx

```diff
@@ -0,0 +1,179 @@
+ * limitations under the License.
+ */
+
+import React, { useState, useCallback, useRef } from 'react';
+import styled from 'styled-components';
+import copyToClipboard from 'design/utils/copyToClipboard';
+import selectElementContent from 'design/utils/selectElementContent';
+import { ButtonSecondary, Box, Flex } from 'design';
+import { Copy, Check } from 'design/Icon';
+
+const ONE_SECOND_IN_MS = 1000;
+
+export function TextSelectCopyMulti({ lines, bash = true }: Props) {
+  const btnRefs = useRef<HTMLElement[]>([]);
+  const [lineEls, setLineEls] = useState<HTMLElement[]>([]);
+
+  const linesRef = useCallback((node: HTMLElement) => {
+    // Grab the mounted line childrens to be able to calculate spacers
+    // for button placements.
+    if (node) {
+      const childElements: HTMLElement[] = Array(node.children.length);
+      for (var i = 0; i < node.children.length; i++) {
+        childElements[i] = node.children[i] as HTMLElement;
+      }
+      setLineEls(childElements);
+    }
+  }, []);
+
+  function onCopyClick(index) {
+    copyToClipboard(lines[index].text).then(() => {
+      const targetEl =
+        btnRefs.current[index].getElementsByClassName('icon-container')[0];
+      targetEl.classList.toggle('copied');
+
+      setTimeout(() => {
+        targetEl.classList.toggle('copied');
+      }, ONE_SECOND_IN_MS);
+    });
+
+    const targetEl = lineEls[index].getElementsByClassName('text-to-copy')[0];
+    selectElementContent(targetEl as HTMLElement);
+  }
+
+  const isFirefox = window.navigator?.userAgent
+    ?.toLowerCase()
+    .includes('firefox');
+
+  return (
+    <Flex
+      bg="bgTerminal"
+      px={3}
+      pt={2}
+      // Firefox does not add space for visible scrollbars
+      // like it does for chrome and safari.
+      pb={isFirefox ? 3 : 2}
+      alignItems="flex-start"
+      justifyContent="space-between"
+      borderRadius={2}
+      css={{
+        // Make it visible to user only when we have buttons to render
+        // to avoid possible flashing between renders.
+        visibility: lineEls.length > 0 ? 'none' : 'hidden',
+      }}
+    >
+      <Lines mr={1} ref={linesRef}>
+        {lines.map((line, index) => {
+          const isLastText = index === lines.length - 1;
+          return (
+            <Box pt={2} pb={isLastText ? 0 : 2} key={index}>
+              {line.comment && <Comment>{line.comment}</Comment>}
+              <Flex key={index}>
+                {bash && <Box mr="1">{`$`}</Box>}
+                <div className="text-to-copy">{line.text}</div>
+              </Flex>
+            </Box>
+          );
+        })}
+      </Lines>
+      <Box>
+        {lineEls.map((el, index) => {
+          const isLastText = index === lines.length - 1;
+          let spacer;
+          if (lines[index].comment) {
+            spacer = (
+              <Box
+                height={(el.firstElementChild as HTMLElement).offsetHeight}
```

Comment:
I think this could be computed without rendering elements. we know:
- a number of lines: `lineText.split("\n").length`
- line hight: `20px` 
- spacing

An alternative solution would be to use pure CSS. something with position absolute/sticky.

Reply:
thanks for giving me the confidence for the position absolute, i didn't think it will work but it did 🎉